Client/src/Remote/FluentRequest.tsx

```diff
@@ -0,0 +1,93 @@
+import SnackbarUtils from "../Components/Utils/SnackbarUtils"
+import { stringify } from "query-string"
+
+const requestBase: RequestInit = {
+  mode: 'cors', // no-cors, *cors, same-origin
+  cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
+  credentials: 'same-origin', // include, *same-origin, omit
+  redirect: 'follow', // manual, *follow, error
+  referrerPolicy: 'no-referrer', // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+}
+
+export const get = (route: string): FluentRequest => {
+  return new FluentRequest('GET', route)
+}
+
+export const post = (route: string): FluentRequest => {
+  return new FluentRequest('POST', route)
+}
+
+export const put = (route: string): FluentRequest => {
+  return new FluentRequest('PUT', route)
+}
+
+//underscore is needed so that JS does not confuse it with reserved delete keyword
+export const _delete = (route: string): FluentRequest => {
+  return new FluentRequest('DELETE', route)
+}
+
+export const patch = (route: string): FluentRequest => {
+  return new FluentRequest('PATCH', route)
+}
+
+export class FluentRequest {
+  private url: string
+  private query: string
+  private requestInit: RequestInit
+  private headers: Headers
```

Comment:
Request init sets http headers, dont think we need to split them here between requestInit and headers

Reply:
![image](https://user-images.githubusercontent.com/22376976/106365355-054e4780-6303-11eb-813c-93354993fe54.png)
no headers on request init, this.headers are put in request init right before the call is done, unless mode, cache etc. are also headers?

Reply:
Yeah they are